src/routes/add-custom-discipline/AddCustomDisciplineScreen.tsx

```diff
@@ -86,34 +87,36 @@ const AddCustomDiscipline = ({ navigation }: AddCustomDisciplineScreenProps): JS
   return (
     <Loading isLoading={loading}>
       {customDisciplines && (
-        <Container>
-          <CustomDisciplineHeading>{labels.addCustomDiscipline.heading}</CustomDisciplineHeading>
-          <Description>{labels.addCustomDiscipline.description}</Description>
-          <InputContainer>
-            <CustomTextInput
-              errorMessage={errorMessage}
-              placeholder={labels.addCustomDiscipline.placeholder}
-              value={code}
-              onChangeText={setCode}
-              rightContainer={
-                <TouchableOpacity onPress={() => setShowQRCodeOverlay(true)}>
-                  <QRCodeIcon accessibilityLabel='qr-code-scanner' width={wp('6%')} height={wp('6%')} />
-                </TouchableOpacity>
-              }
+        <RouteWrapper>
```

Comment:
Shouldn't this be the outermost component? I.e. wrapped around loading?

Reply:
you are right